tools/src/code-review/index.ts

```diff
@@ -0,0 +1,139 @@
+import chalk from 'chalk';
+
+import Git from '../Git';
+import * as GitHub from '../GitHub';
+import logger from '../Logger';
+import { generateReviewBodyFromOutputs } from './reports';
+import checkMissingChangelogs from './reviewers/checkMissingChangelogs';
+import { ReviewEvent, ReviewComment, ReviewInput, ReviewOutput, ReviewStatus } from './types';
+
+/**
+ * An array with functions whose purpose is to check and review the diff.
+ */
+const REVIEWERS = [checkMissingChangelogs];
+
+/**
+ * Goes through the changes included in given pull request and checks if they meet basic requirements.
+ */
+export async function reviewPullRequestAsync(prNumber: number) {
+  const pr = await GitHub.getPullRequestAsync(prNumber);
+  const user = await GitHub.getAuthenticatedUserAsync();
+
+  logger.info('👾 Fetching base commit:', chalk.yellow.bold(pr.base.sha));
+  await Git.fetchAsync({
+    remote: 'origin',
+    ref: pr.base.sha,
+  });
+
+  logger.info('👾 Fetching head commit:', chalk.yellow.bold(pr.head.sha));
+  await Git.fetchAsync({
+    remote: 'origin',
+    ref: pr.head.sha,
+  });
+
+  // Find the common ancestor of the base and PR's head.
+  const mergeBaseSha = await Git.mergeBaseAsync(pr.base.sha, pr.head.sha);
+  logger.info('👀 Found common ancestor:', chalk.yellow.bold(mergeBaseSha));
+
+  // Gets the diff of the pull request.
+  const diff = await Git.getDiffAsync(mergeBaseSha, pr.head.sha);
+
+  const input: ReviewInput = {
+    pullRequest: pr,
+    mergeBaseSha,
+    diff,
+  };
+
+  // Run all the checks asynchronously and collects their outputs.
+  logger.info('🕵️‍♀️  Reviewing changes');
+  const outputs = (await Promise.all(REVIEWERS.map((reviewer) => reviewer(input)))).filter(
+    Boolean
+  ) as ReviewOutput[];
+
+  // Only active (non-passive) outputs will be reported in the review body.
+  const activeOutputs = outputs.filter(
+    (output) => output.title && output.body && output.status > ReviewStatus.PASSIVE
```

Comment:
```suggestion
    (output) => output.title && output.body && output.status !== ReviewStatus.PASSIVE
```
personally i prefer `!==` here, as the significance of `> someEnum` isn't instantly understandable